internal/gamepaddb/gen.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+
+// This program generates platform-specific controller dbs. It can be invoked by running
+// go generate
+package main
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+const license = `// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+`
+
+// gamecontrollerdb.txt is downloaded at https://github.com/mdqinc/SDL_GameControllerDB.
+
+// To update the database file, run:
+//
+//     curl --location --remote-name https://raw.githubusercontent.com/mdqinc/SDL_GameControllerDB/master/gamecontrollerdb.txt
+
+//go:embed gamecontrollerdb.txt
+var gamecontrollerdb_txt []byte
+
+const db_template string = `
+{{.License}}
```

Comment:
Snake cases are not a convention in Ebitengine. Rename `db_template` to `dbTemplate`. Also, remove the first new line.

```
const dbTemplate = `{{.License}}
```